backend/infrahub/core/account.py

```diff
@@ -145,24 +145,93 @@ async def query_init(self, db: InfrahubDatabase, **kwargs: Any) -> None:
         WITH account, r1 as r
         WHERE r.status = "active"
         WITH account
-        MATCH group_path = (account)-[]->(:Relationship {name: "group_member"})
-            <-[]-(:%(account_group_node)s)
-            -[]->(:Relationship {name: "role__accountgroups"})
-            <-[]-(:%(account_role_node)s)
-            -[]->(:Relationship {name: "role__permissions"})
-            <-[]-(object_permission:%(object_permission_node)s)
-            -[:HAS_ATTRIBUTE]->(:Attribute {name: "branch"})
-            -[:HAS_VALUE]->(object_permission_branch:AttributeValue)
```

Comment:
I broke this query up into 4 subqueries to correctly handle deletes along each step of the path. now it is one subquery per relationship, (so account -> account group, account_group -> account_role, etc.)